packages/rocketchat-api/server/v1/users.js

```diff
@@ -95,6 +95,25 @@ RocketChat.API.v1.addRoute('users.getAvatar', { authRequired: false }, {
 	},
 });
 
+RocketChat.API.v1.addRoute('users.setActiveStatus', { authRequired: true }, {
+	post() {
+		check(this.bodyParams, {
+			userId: String,
+			activeStatus: Boolean,
+		});
+
+		if (!RocketChat.authz.hasPermission(this.userId, 'edit-other-user-info')) {
```

Comment:
With this verification, a user that doesn't have the `edit-other-user-info` permission would *not* be able to set their own active status. Maybe you should also verify that :)

Reply:
Can I do like this [endpoint](https://github.com/RocketChat/Rocket.Chat/blob/78b41a3211fa6a312cc689397e3aaf52ef32c014/packages/rocketchat-api/server/v1/users.js#L235)?

Reply:
I tried to do this, but throws the **error-not-allowed**, because the method `setUserActiveStatus` allows only the users who have the [`edit-other-user-active-status`](https://github.com/RocketChat/Rocket.Chat/blob/d25d147d745b94dcf2bda366edf9ab5002e066a4/server/methods/setUserActiveStatus.js#L18) permission.


By the way, I put the wrong permission, I'll fix it.

Reply:
Sorry, I misunderstood this method. I thought it was supposed to change the _user's presence_, but instead, it changes the _status_, which actually controls if the user is able to login on the system.

So, considering the actual method's purpose, this is the expected behavior.

I have another consideration to make, though. The method you are calling already checks for the desired permission, so I think you could wrap the call in a `try ... catch` block to send the unauthorized response.

Does it make sense?

Reply:
Yes, makes sense. I'll do it.

Reply:
@d-gubert I think the `try ... catch` is not necessary, because the api already wrap all the endpoints in a `try ... catch` as we can [see here](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-api/server/api.js#L221).

Reply:
@MarcosSpessatto Should I still check the permission?

Reply:
@thayannevls yes, you can check.